.circleci/config.yml

```diff
@@ -1,6 +1,86 @@
 version: 2.1
 
 commands:
+  conditionally_exclude_changes:
+    description: 'Stop the job when there are no active changes in the branch. However, the master branch not affected.'
+    parameters:
+      exclude:
+        description: 'Prevent the test from stopping if changes are found in this directory'
+        type: string
+    steps:
+      - run:
+          name: Require any directory except << parameters.exclude >> to have changes
+          command: |
+            ROOT_BRANCH="master"
+
+            if [ $CIRCLE_BRANCH = $ROOT_BRANCH ]; then
+              exit 0
+            fi
+
+            ROOT_COMMIT=$(git merge-base remotes/origin/"${CIRCLE_BRANCH}" remotes/origin/"${ROOT_BRANCH}")
+            echo "Checking for changes in branch ${CIRCLE_BRANCH}"
+            echo " - Excluding << parameters.exclude >>"
+            echo " - Git root ${CIRCLE_WORKING_DIRECTORY}/*"
+            echo " - Initial commit ${ROOT_COMMIT}"
+            # ROOT_COMMIT=$(git merge-base remotes/origin/"${CIRCLE_BRANCH}" "${ROOT_BRANCH}")
+            # Tests all root directories
+            for file in $PWD/*/
+            do
+              FILE_NAME=$(basename $file)
+              if [[ ! " << parameters.exclude >> " =~ " ${FILE_NAME} " ]]; then
```

Comment:
Get the path components precise so that we can just compare with `=` instead of the regex match

Reply:
this could allow for an array in the future